include/openthread/types.h

```diff
@@ -994,26 +994,6 @@ typedef struct
     uint16_t        mSecretKeyLength;    ///< Secret key length in bytes. Should be at least 16 bytes == 128 bits.
 } otSemanticallyOpaqueIidGeneratorData;
 
-/**
- * ICMPv6 Message Types
- *
- */
-typedef enum otIcmp6Type
-{
-    kIcmp6TypeDstUnreach  = 1,     ///< Destination Unreachable
-    kIcmp6TypeEchoRequest = 128,   ///< Echo Request
```

Comment:
Currently we expose API to register ICMPv6 handler in application space. Thus those types are very useful (in `include/types.h` or `include/icmp6.h`)

E.g. Application may act on specific types without defining them by its own:

```
//main.c
m_icmp6_handler.mReceiveCallback = icmp_receive_callback;
otIcmp6RegisterHandler(p_instance, &m_icmp6_handler);

static void icmp_receive_callback(void * p_context, otMessage * p_message,
                                  const otMessageInfo * p_message_info, const otIcmp6Header * p_icmp_header)
{
    if (p_icmp_header->mType == kIcmp6TypeEchoRequest)
    {
        // Do sth..
    }
}
```